test/general-tests.js

```diff
@@ -15,26 +15,67 @@ describe('General Tests', () => {
 		assert.equal(typeof FuelAuth, 'function');
 	});
 
+	it('clientSecret not needed for OAuth2 public app', () => {
+		let AuthClient;
+		
+		var options = {
+			clientId:'client_id',
+			authUrl:'test',
+			authOptions:{
+				authVersion: 2,
+				applicationType: 'public',
+				redirectURI: 'test',
+				authorizationCode: 'test'
+			}
+		};
+		AuthClient = new FuelAuth(options);
+		assert(true);
+	});
+
+
+	it('AuthorizationCode and RedirectURI are mandatory for public/web app', () => {
```

Comment:
Using the following type of assertion, we do not need the assert.fail and we can assert on multiple properties of the exception thrown (name, message .. )

```
assert.throws(()=>{new FuelAuth(options);},
	{
		name: 'Error',
		message: 'RedirectURI and Authorization Code are required for Public/Web App OAuth2 Authentication'
	}
);
```
Also, I think we should test all the four combinations ( ( public, web ) X ( redirectURI, authorizationCode ) )